### ISSUE TITLE
[Bug] Can't trade with travelling wizards or normal wizards

### ISSUE THREAD

**Issue by RatseerOfRattesse** [WinDanesz/AncientSpellcraft#155]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Observed behaviour

I summoned a travelling wizard with a bartering scroll. When I tried to trade with him, I shift clicked the trade, and it gave me on item and got sold out, but the item disappeared when I exited the trading GUI and my payment was refunded. When I tried to trade with the wizard normally, without shift-clicking, that trade became sold out and nothing entered my inventory. My payment disappeared, but reappeared when I exited the trading GUI. When I clicked on the wizard again, he was no longer sold out. I've tried re-logging into my world and also reloading Minecraft.

### Expected behaviour

Shift-clicking should have taken all of the wizards stock, trading without shift-clicking should have boughten one item.

### Steps to reproduce

1. Join singleplayer world
2. Use a scroll of bartering
3. Attempt to shift-click trade or normal trade the summoned wizard.

### Environment (Singleplayer/Server, etc.)

Singleplayer

### Mod version

1.5.10

### Forge version

14.23.5.2859

### Other mods

Haven't listed mods that are not connected to the error whatsoever, as it's a 140 mod custom modpack.
**Possibly Relevant Mods**

CompatSkills 1.16.3 (later version limits arcane workbench use, not sure about this version)
Arcane Essentials 1.0.1
Baubles 1.5.2
EBW 4.3.11
EBW Twilight Forest Spell Pack 1.1.0
EBW Extra Spells 1.2.0

(Taken from latest crash report. That report was not relevant to the issue, I just don't know which mods are coremods)
**Coremods**
  ParticleCullingLoadingPlugin (particleculling-1.12.2-v1.4.1.jar)
  PCF Core Mod (PhosphorCrashFix-1.12.2-1.0.0.jar)
  Quark Plugin (Quark-r1.6-179.jar)
  ObfuscatePlugin (obfuscate-0.4.2-1.12.2.jar)
  iceandfire (iceandfire-1.9.1-1.12.2.jar)
  EntityCullingPlugin (EntityCulling-1.12.2-6.4.1.jar)
  EnderCorePlugin (EnderCore-1.12.2-0.5.76-core.jar)
  PhosphorFMLLoadingPlugin (phosphor-1.12.2-0.2.6+build50-universal.jar)
  ratscore (rats-3.2.14-1.12.2.jar)
  SpartanWeaponry-MixinLoader (SpartanWeaponry-1.12.2-1.5.3.jar)
  IvToolkit (IvToolkit-1.3.3-1.12.jar)
  LoadingPlugin (Reskillable-1.12.2-1.13.0.jar)
  RenderLibPlugin (RenderLib-1.12.2-1.3.1.jar)
  RandomPatches (randompatches-1.12.2-1.22.1.10.jar)
  TheBetweenlandsLoadingPlugin (TheBetweenlands-3.9.6-core.jar)
  (foamfix-0.10.15-1.12.2.jar)
  SoManyEnchantments ASM (SoManyEnchantments-0.5.5.jar)
  ForgelinPlugin (Forgelin-1.8.4.jar)
  FutureMC (future-mc-0.2.11.jar)
  ApotheosisCore (Apotheosis-1.12.2-1.12.5.jar)
  CTMCorePlugin (CTM-MC1.12.2-1.0.2.31.jar)
  LucraftCoreCoreMod (LucraftCore-1.12.2-2.4.17.jar)
  SecurityCraftLoadingPlugin ([1.12.2] SecurityCraft v1.9.7.jar)
  llibrary (llibrary-core-1.0.11-1.12.2.jar)
  AstralCore (astralsorcery-1.12.2-1.10.27.jar)

**Comment by WinDanesz** [comment]
Does this happen with regular click trades too, without shift clicking?

**Comment by anabsolutesloth** [comment]
This is an issue with FutureMC's modernized trade GUI, it happens with base EBW aswell and is present in MC Eternal, inputting the second item manually will work around the issue.

**Comment by WinDanesz** [comment]
IIRC there is a setting in future mc to disable the trading gui changes

**Comment by RatseerOfRattesse** [comment]
Yes, it does happen with regular clicks. I'll take a look at the FutureMC as I've been using that.

Update: Yeah, it works. Thanks.

**Comment by KnightCa** [comment]
Would be nice to have EBWizardry be compatible with Future MC's modernized trade GUI (since it's widely used).
